go/parquet/encryption_read_config_test.go

```diff
@@ -0,0 +1,443 @@
+			rowsRead, valuesRead, _ = boolReader.ReadBatch(1, val[:], nil, nil)
+			// ensure only 1 value is read
+			d.EqualValues(1, rowsRead)
+			// there are no null values
+			d.EqualValues(1, valuesRead)
+			// verify the value
+			expected := i%2 == 0
+			d.Equal(expected, val[0], "i: ", i)
+			i++
+		}
+		d.EqualValues(i, boolMd.NumValues())
+
+		// Get column reader for int32 column
+		colReader = rowGroupReader.Column(1)
+		int32reader := colReader.(*file.Int32ColumnChunkReader)
+
+		int32md, _ := rgMeta.ColumnChunk(1)
+		// Read all rows in column
+		i = 0
+		for int32reader.HasNext() {
+			var val [1]int32
+			// read one value at a time. the number of rows read is returned. values
+			// read contains the number of non-null rows
+			rowsRead, valuesRead, _ = int32reader.ReadBatch(1, val[:], nil, nil)
+			// ensure only 1 value is read
+			d.EqualValues(1, rowsRead)
+			// there are no null values
+			d.EqualValues(1, valuesRead)
+			// verify the value
+			d.EqualValues(i, val[0])
+			i++
+		}
+		d.EqualValues(i, int32md.NumValues())
+
+		// Get column reader for int64 column
+		colReader = rowGroupReader.Column(2)
+		int64reader := colReader.(*file.Int64ColumnChunkReader)
+
+		int64md, _ := rgMeta.ColumnChunk(2)
+		// Read all rows in column
+		i = 0
+		for int64reader.HasNext() {
+			var (
+				val [1]int64
+				def [1]int16
+				rep [1]int16
+			)
+
+			// read one value at a time. the number of rows read is returned. values
+			// read contains the number of non-null rows
+			rowsRead, valuesRead, _ = int64reader.ReadBatch(1, val[:], def[:], rep[:])
+			// ensure only 1 value is read
+			d.EqualValues(1, rowsRead)
+			// there are no null values
+			d.EqualValues(1, valuesRead)
+			// verify the value
+			expectedValue := int64(i) * 1000 * 1000 * 1000 * 1000
+			d.Equal(expectedValue, val[0])
+			if i%2 == 0 {
+				d.EqualValues(1, rep[0])
+			} else {
+				d.Zero(rep[0])
+			}
+			i++
+		}
+		d.EqualValues(i, int64md.NumValues())
+
+		// Get column reader for int96 column
+		colReader = rowGroupReader.Column(3)
+		int96reader := colReader.(*file.Int96ColumnChunkReader)
+
+		int96md, _ := rgMeta.ColumnChunk(3)
+		// Read all rows in column
+		i = 0
+		for int96reader.HasNext() {
+			var (
+				val [1]parquet.Int96
+			)
+
+			// read one value at a time. the number of rows read is returned. values
+			// read contains the number of non-null rows
+			rowsRead, valuesRead, _ = int96reader.ReadBatch(1, val[:], nil, nil)
+			// ensure only 1 value is read
+			d.EqualValues(1, rowsRead)
+			// there are no null values
+			d.EqualValues(1, valuesRead)
+			// verify the value
+			var expectedValue parquet.Int96
+			binary.LittleEndian.PutUint32(expectedValue[:4], uint32(i))
+			binary.LittleEndian.PutUint32(expectedValue[4:], uint32(i+1))
+			binary.LittleEndian.PutUint32(expectedValue[8:], uint32(i+2))
+			d.Equal(expectedValue, val[0])
+			i++
+		}
+		d.EqualValues(i, int96md.NumValues())
+
+		if decryptConfigNum != 3 {
```

Comment:
i think it is worth commenting here and above why this logic doesn't apply to case number 3

Reply:
added a comment